### PR TITLE
fix: mimic the pubsub test to clean up bigquery datasets

### LINKED ISSUE #635: Usage guide test leaving BQ leftovers
Every time the samples usage_guide test runs it leaves a [leftover dataset](https://screenshot.googleplex.com/8uH7AkhnTeFNiDH) in BigQuery. This test needs to be refactored to tear down the BigQuery dataset upon test completion. https://github.com/googleapis/python-logging/blob/330b53c26dd4b653603128bf815591742c818098/samples/snippets/usage_guide_test.py#L51-L58

### PR DESCRIPTION
**Pull request by leahecole**
This should fix #635

### CODE REVIEW

**Comment by leahecole** [comment]
Tests are failing on a quota error, I'll try to rerun

**Comment by leahecole** [comment]
Looking at the project in question and am not seeing anything near the quota limit [screenshot](https://screenshot.googleplex.com/4CXxgud3Kxocx9N) 🤔

**Review by daniel-sanche — APPROVED** [review]

File: samples/snippets/usage_guide.py [hunk 1]
```diff
@@ -315,13 +315,18 @@ def _sink_bigquery_setup(client):
     client.update_dataset(dataset, ["access_entries"])  # API call
     # [END sink_dataset_permissions]
 
-    return dataset
+    class DatasetDeleter:
```
Comment (dandhlee) [inline samples/snippets/usage_guide.py:318]: nit: Could you add a one line comment to explain what it does?

**Review by dandhlee — APPROVED** [review]
LGTM, one minor nit.

**Pull request merged**